train/train.py

```diff
@@ -1,39 +1,146 @@
+                           Must match the name(s) of the data directories.
+        models (list<str>): Comma-separated list of model names to train.
+                            Options: ['rf', 'ann', 'lasso', 'xgb']
+        predictors (list<str>): Comma-separated list of predictors. Ignored if
+                                predictors_path is provided.
+        predictors_paths (list<str>): Comma-separated list path file(s) with
+                                      predictor names. See train/predictors.txt
+                                      for an example.
+        log_metrics (list<str>): Validation metrics to log.
+        inner_cv (int): Number of folds for k-fold cross validation in the
+                        training set(s) for selecting model hyperparameters.
+        n_iter (int): Number of parameter settings that are sampled in the
+                      inner cross validation.
+        overwrite_existing_models (bool): Whether to overwrite models if they
+                                          already exist.
 
     Saves trained models to data/{SiteID}/{model}/{predictor}/
     for each {SiteID}, {model}, and {predictor} subset.
     """
     data_dir = Path("data/")
-    sites = sites.split(",")
-    models = models.split(",")
+    if isinstance(sites, str):
+        sites = sites.split(",")
+    if isinstance(models, str):
+        models = models.split(",")
+
+    if predictors_paths is not None:
+        if isinstance(predictors_paths, str):
+            predictors_paths = predictors_paths.split(",")
+        predictor_subsets = {}
+        for predictors_path in predictors_paths:
+            predictors_path = Path(predictors_path)
+            with predictors_path.open() as f:
+                predictor_subset = f.read().splitlines()
+            predictor_subsets[predictors_path.stem] = predictor_subset
+    elif predictors is not None:
+        if isinstance(predictors, str):
+            predictors = predictors.split(",")
+        predictor_subsets = {"predictors": predictors}
+    else:
+        raise ValueError("Must provide predictors or predictors_paths.")
+
     for site in sites:
+        # Ensure that preprocessing has been run on this site
         site_data_dir = data_dir / site
-        for predictor in predictors:
-            # TODO: If train/valid/test csvs don't exist, tell user to run preprocessing
-            # TODO: Add data munging for input to ML here
-            for model in models:
-                print(f"Training model {model} for site {site}...")
-                model_dir = site_data_dir / model / predictor
-                model_dir.mkdir(exist_ok=True)
-                for i in range(n_train):
-                    model_path = model_dir / f"model{i+1}.pkl"
+        train_dir = site_data_dir / "training"
+        if not train_dir.exists():
+            raise ValueError(f"You must run preprocessing on site {site}" + 
+                             "before training.")
+        
```

Comment:
```suggestion
        if len(os.listdir(train_dir)) < 1:
            raise ValueError(f"You must run preprocessing on site {site}" +
                                      "before training.")
```

Reply:
Thanks for the suggestion - implemented.